src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -323,23 +331,48 @@ public BlobStoreIndexShardSnapshot loadSnapshot() {
         }
 
         /**
-         * Removes all unreferenced files from the repository
+         * Removes all unreferenced files from the repository and writes new index file
          *
          * @param snapshots list of active snapshots in the container
          * @param blobs     list of blobs in the container
          */
-        protected void cleanup(List<BlobStoreIndexShardSnapshot> snapshots, ImmutableMap<String, BlobMetaData> blobs) {
+        protected void finalize(List<SnapshotFiles> snapshots, int fileListGeneration, ImmutableMap<String, BlobMetaData> blobs) {
             BlobStoreIndexShardSnapshots newSnapshots = new BlobStoreIndexShardSnapshots(snapshots);
+            String newSnapshotIndexName = SNAPSHOT_INDEX_PREFIX + fileListGeneration;
+            // If we deleted all snapshots - we don't need to create the index file
+            if (snapshots.size() > 0) {
+                try (OutputStream output = blobContainer.createOutput(SNAPSHOT_TEMP_PREFIX + fileListGeneration)) {
+                    XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON, output);
```

Comment:
The snapshot file can be compressed, do you think this snapshot index file should be compressed too?